src/main/java/org/reaktivity/specification/mqtt/internal/MqttFunctions.java

```diff
@@ -0,0 +1,291 @@
+        {
+            beginExRW.typeId(typeId);
+            return this;
+        }
+
+        public MqttBeginExBuilder clientId(
+           String clientId)
+        {
+            beginExRW.clientId(clientId);
+            return this;
+        }
+
+        public MqttBeginExBuilder topic(
+           String topic,
+           int options)
+        {
+            beginExRW.topicsItem(a -> a.topic(topic).options(options));
+            return this;
+        }
+
+        public MqttBeginExBuilder packetId(
+           int packetId
+        )
+        {
+            beginExRW.packetId(packetId);
+            return this;
+        }
+
+        public MqttBeginExBuilder reason(
+           int reason)
+        {
+            beginExRW.reasonCodesItem(b -> b.reasonCode(reason));
+            return this;
+        }
+
+        public MqttBeginExBuilder role(
+           String role)
+        {
+            beginExRW.role(p -> p.set(MqttRole.valueOf(role)));
+            return this;
+        }
+
+        public byte[] build()
+        {
+            final MqttBeginExFW beginEx = beginExRW.build();
+            final byte[] array = new byte[beginEx.sizeof()];
+            beginEx.buffer().getBytes(beginEx.offset(), array);
+            return array;
+        }
+    }
+
+    public static final class MqttDataExBuilder
+    {
+        private final MqttDataExFW.Builder dataExRW;
+
+        private MqttDataExBuilder()
+        {
+            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[1024 * 8]);
+            this.dataExRW = new MqttDataExFW.Builder().wrap(writeBuffer, 0, writeBuffer.capacity());
+        }
+
+        public MqttDataExBuilder typeId(
+           int typeId)
+        {
+            dataExRW.typeId(typeId);
+            return this;
+        }
+
+        public MqttDataExBuilder packetId(
+           int packetId
+        )
+        {
+            dataExRW.packetId(packetId);
+            return this;
+        }
+
+        public MqttDataExBuilder topic(
+           String topic)
+        {
+            dataExRW.topic(topic);
+            return this;
+        }
+
+        public MqttDataExBuilder messageExpiry(
+           long msgExp)
+        {
+            dataExRW.messageExpiry(msgExp);
+            return this;
+        }
+
+        public MqttDataExBuilder contentType(
+           String contentType)
+        {
+            dataExRW.contentType(contentType);
+            return this;
+        }
+
+        public MqttDataExBuilder payloadFormat(
+           String format)
+        {
+            dataExRW.payloadFormat(p -> p.set(MqttPayloadFormat.valueOf(format)));
+            return this;
+        }
+
+        public MqttDataExBuilder respTopicInfo(
+           String topic,
+           String correlationData)
+        {
+            dataExRW.respTopicInfo(r -> r.topic(topic)
+                    .correlationInfo(c -> c.bytes(b -> b.set(correlationData.getBytes()))));
+            return this;
```

Comment:
Suggest keeping a one-to-one relationship between EL builder methods and flyweight builder methods.